### ISSUE TITLE
old python version causes urllib3 certs error

### ISSUE THREAD

**Issue by hacpai** [wakatime/vim-wakatime#21]
```
➜ /Users/MasterMac >Traceback (most recent call last):
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/cli.py", line 15, in <module>
    import wakatime
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/__init__.py", line 17, in <module>
    from .base import main
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/base.py", line 38, in <module>
    from .packages import requests
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/__init__.py", line 58, in <module>
    from . import utils
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/utils.py", line 26, in <module>
    from .compat import parse_http_list as _parse_list_header
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/compat.py", line 42, in <module>
    from .packages.urllib3.packages.ordered_dict import OrderedDict
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/packages/__init__.py", line 95, in load_module
    raise ImportError("No module named '%s'" % (name,))
ImportError: No module named 'wakatime.packages.requests.packages.urllib3'
```

**Comment by alanhamlett** [comment]
I just updated the included requests package to v2.6.0, hopefully this fixes it. Can you run `:VundleUpdate` inside Vim to update the vim-wakatime plugin and see if it works now?

**Comment by arcticlion** [comment]
@alanhamlett 
I update Waketime, but it still error.

```
 File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/base.py", line 38, in <module>
                                                         from .packages import requests
         File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/__init__.py", line 58, in <module>
                                                          from . import utils
                                                                               File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/utils.py", line 26, in <module>
                                             from .compat import parse_http_list as _parse_list_header
                        File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/compat.py", line 42, in <module>
                                                                       from .packages.urllib3.packages.ordered_dict import OrderedDict
                                                        File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/packages/__init__.py", line 95, in load_module
                                     raise ImportError("No module named '%s'" % (name,))
        ImportError: No module named 'wakatime.packages.requests.packages.urllib3'
```

**Comment by alanhamlett** [comment]
I've added `wakatime.packages.requests.packages` to `sys.path` so requests should be able to import `urllib3` now. Can you try updating vim-wakatime again?

**Comment by arcticlion** [comment]
@alanhamlett It still don't work. How can I help you?

```
Traceback (most recent call last):
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/cli.py", line 15, in <module>
    import wakatime
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/__init__.py", line 17, in <module>
    from .base import main
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/base.py", line 39, in <module>
    from .packages import requests


  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/__init__.py", line 58, in <module>
    from . import utils
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/utils.py", line 26, in <module>
    from .compat import parse_http_list as _parse_list_header
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/compat.py", line 42, in <module>
    from .packages.urllib3.packages.ordered_dict import OrderedDict
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/packages/__init__.py", line 95, in load_module
    raise ImportError("No module named '%s'" % (name,))
ImportError: No module named 'wakatime.packages.requests.packages.urllib3'
```

**Comment by alanhamlett** [comment]
Let's try a new approach. Can you open a Terminal and type:

    cd /Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/
    python
    import wakatime.packages.requests.packages.urllib3

Hopefully that produces an exception.

**Comment by arcticlion** [comment]
@alanhamlett 
```
>>> import wakatime.packages.requests.packages.urllib3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/__init__.py", line 17, in <module>
    from .base import main
  File "/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/base.py", line 39, in <module>
    from .packages import requests
  File "wakatime/packages/requests/__init__.py", line 58, in <module>
    from . import utils
  File "wakatime/packages/requests/utils.py", line 25, in <module>
    from . import certs
ImportError: cannot import name certs
```

**Comment by alanhamlett** [comment]
Do both these files exist?

`/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/cacert.pem`
`/Users/MasterMac/.vim/bundle/vim-wakatime/plugin/packages/wakatime/packages/requests/certs.py`

**Comment by arcticlion** [comment]
@alanhamlett Yes, these files exist.

This is my certs.py

```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-

"""
certs.py
~~~~~~~~

This module returns the preferred default CA certificate bundle.

If you are packaging Requests, e.g., for a Linux distribution or a managed
environment, you can change the definition of where() to return a separately
packaged CA bundle.
"""
import os.path

try:
    from certifi import where
except ImportError:
    def where():
        """Return the preferred certificate bundle."""
        # vendored bundle inside Requests
        return os.path.join(os.path.dirname(__file__), 'cacert.pem')

if __name__ == '__main__':
    print(where())
```

**Comment by alanhamlett** [comment]
Can you try the latest version of vim-wakatime? Sorry about this.

It's unrelated to your version of Python, because I've just tested importing `wakatime.packages.requests.packages.urllib3` with Python2.6.9, Python2.7.6, and Python 2.7.9 without error.

This is very strange... I've added `wakatime.packages.requests.packages.urllib3` and `wakatime.packages.requests.packages.urllib3.util` to `sys.path` and crossing my fingers.

**Comment by arcticlion** [comment]
@alanhamlett Yeah, this is very strange. My vim-wakatime is installed through Vundle and run :VundleUpdate inside Vim to update the vim-wakatime plugin. But it still doesn't work.

**Comment by alanhamlett** [comment]
Can you try `sudo pip install --upgrade requests urllib3`

This might get it working for you, but doesn't help us solve the underlying problem.

**Comment by arcticlion** [comment]
@alanhamlett I upgrade python to 2.7.9. vim-wakatime. Thank you for your help.

**Comment by alanhamlett** [comment]
Oh so it was that version of Python?

**Comment by arcticlion** [comment]
python 2.9.7
-- 
Henry
Sent with Airmail

在 2015年4月2日 下午3:05:45, Alan Hamlett (notifications@github.com) 写到:

Oh so it was that version of Python?

**Comment by alanhamlett** [comment]
ok, so it's working?

**Comment by arcticlion** [comment]
Yeah, there is no longer appear exception.
-- 
Henry
Sent with Airmail

在 2015年4月2日 下午3:08:00, Alan Hamlett (notifications@github.com) 写到:

ok, so it's working?

**Comment by alanhamlett** [comment]
Good! Let's leave this open so I can fix the problem for others.

**Comment by arcticlion** [comment]
Okay ;-)
-- 
Henry
Sent with Airmail

在 2015年4月2日 下午3:16:29, Alan Hamlett (notifications@github.com) 写到:

Good! Let's leave this open so I can fix the problem for others.

**Comment by alanhamlett** [comment]
Fixed with wakatime/wakatime@df7bf5cddbc6fb5cddbecb2768268cb0cfde38fe.
